src/shared/pager.c

```diff
@@ -171,7 +171,7 @@ int pager_open(PagerFlags flags) {
                  * pager. If they didn't, use secure mode when under euid is changed. If $SYSTEMD_PAGERSECURE
                  * wasn't explicitly set, and we autodetect the need for secure mode, only use the pager we
                  * know to be good. */
-                int use_secure_mode = getenv_bool("SYSTEMD_PAGERSECURE");
+                int use_secure_mode = getenv_bool_secure("SYSTEMD_PAGERSECURE");
```

Comment:
Yeah, makes sense. If somebody comes up with a case where this break some legitimate use case, we can always consider relaxing the rule. But it's better to start out strict.